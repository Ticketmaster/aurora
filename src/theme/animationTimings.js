const listTimings = {
  caretRotate: 100,
  collapse: 300,
  contentFadeOut: 100,
  contentFadeIn: 300,
  contentFadeInDelay: 100
};

export default listTimings;
