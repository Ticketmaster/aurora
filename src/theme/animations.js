import { css } from "styled-components";
import CONSTANTS from "./constants";

const { easing: EASING } = CONSTANTS;

const borderRadius = css`margin-bottom 0.3s ${EASING.easeInOutQuad} 0s`
const boxShadow = css`box-shadow 0.3s ${EASING.easeInOutQuad} 0s`;
const marginBottom = css`margin-bottom 0.3s ${EASING.easeInOutQuad} 0s`;
const maxHeight = css`max-height 0.3s ${EASING.easeInOutQuad} 0s`;
const opacity = css`opacity 0.3s cubic-bezier(0.455, 0.03, 0.515, 0.955) 0.2s`;

const accordionSectionText = css`
  .collapse {
    opacity: 0;
    transition: opacity 0.1s cubic-bezier(0.55, 0.085, 0.68, 0.53) 0s;
    visibility: hidden;
  }
  .expand {
    opacity: 1;
    transition: opacity 0.3s cubic-bezier(0.455, 0.03, 0.515, 0.955) 0.2s;
    visibility: visible;
  }
`;

const minTransitionHeight = css`
  height: auto;
  max-height: 300px;
  transition: ${maxHeight};
`;

const maxTransitionHeight = css`
  max-height: 600px;
  transition: ${maxHeight};
`

const adjustHeight = css`
   .minHeight { ${minTransitionHeight}}
   .maxHeight { ${maxTransitionHeight} }
`;

const collapse = css`
  max-height: 0;
  opacity: 0;
  visibility: collapse;
  transition: ${maxHeight}, ${opacity}, ${marginBottom};
`;
const expand = css`
  max-height: 600px;
  opacity: 1;
  visibility: visible;
  transition: ${maxHeight}, ${opacity}, ${marginBottom};
`;

const expandCollapse = css`
  .expand { ${expand} }
  .collapse { ${collapse} }
`;

const hover = css`
  background-color: rgba(2, 108, 223, 0.1);
  -webkit-transition: background-color 100ms linear;
  -ms-transition: background-color 100ms linear;
  transition: background-color 100ms linear;
`;

export {
  adjustHeight,
  accordionSectionText,
  borderRadius,
  boxShadow,
  collapse,
  expand,
  expandCollapse,
  hover,
  marginBottom,
  maxHeight,
  opacity
};
