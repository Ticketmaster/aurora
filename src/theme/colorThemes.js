const success = "#1bab1e";

const global = {
  brand: "#026cdf",
  primary: {
    // brand colors
    base: "#026cdf",
    reverse: "#3396ff",
    reverseLight: "#80b5ef",
    reverseLightest: "#bfdaf6",

    // button colors
    dark: "#0150a7",
    darkest: "#013670",
    light: "#d6e7fa",
    lightest: "#ebf4fd",

    // not used in new components. kept for backward compatibility.
    lightBase: "rgba(2, 108, 223, 0.1)",
    medium: "#0150a7",
    muted: "#d6e7fa"
  },
  special: {
    base: success,
    dark: "#23702c",
    darkest: "#265333",

    medium: "#23702c" // not used in new components. kept for backward compatibility.
  },
  darkFill: "#1f262d", // not used in new components. kept for backward compatibility.
  dark: {
    base: "#1f262d",
    light: "#8f9296",
    lightest: "#e3e4e5"
  },
  accent01: {
    base: "#c56bff",
    dark: "#904eba",
    light: "#eacbff",
    lightest: "#f9f0ff"
  },
  accent02: {
    base: "#962d94",
    dark: "#6d236c",
    light: "#dfc0df",
    lightest: "#f4eaf4"
  },
  accent03: {
    base: "#14a1a3",
    dark: "#0a7e80",
    light: "#aadddd",
    lightest: "#cef2f2",
    reverse: "#20c4c7"
  },
  accent04: "#00ffff",
  gray01: "#262626",
  gray02: "#999999",
  gray03: "#bfbfbf",
  gray04: "#ebebeb",
  gray05: "#f6f6f6",
  error: {
    base: "#d93a3a",
    dark: "#a22b2b",
    light: "#f3c3c3",
    lightest: "#fbebeb"
  },
  caution: {
    base: "#f2bd2a",
    dark: "#c69a22",
    light: "#fae7b4",
    lightest: "#fdf5df"
  },
  success: {
    base: success,
    dark: "#148016",
    light: "#bae5bb",
    lightest: "#e8f6e8"
  },
  transparent: "transparent",
  onyx: {
    base: "rgba(38, 38, 38, 1)",
    light: "rgba(38, 38, 38, 0.65)",
    muted: "rgba(38, 38, 38, 0.4)"
  },
  white: {
    base: "rgba(255, 255, 255, 1)",
    lighter: "rgba(255, 255, 255, 0.2)",
    light: "rgba(255, 255, 255, 0.7)",
    muted: "rgba(255, 255, 255, 0.5)"
  }
};

const themes = {
  global,
  tm: global,
  lne: {
    ...global,
    brand: "#e11d39",
    primary: {
      base: "#2cb3ad",
      medium: "#239893",
      dark: "#1f8783",
      light: "#e9f7f6",
      lightest: "#e9f7f6",
      muted: "#d5f0ef"
    }
  }
};

export default themes;
