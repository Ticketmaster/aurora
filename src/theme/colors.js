import themes from "./colorThemes";

const { tm } = themes;

const colors = {
  // Deprecated colors. Should be removed
  azure: {
    dark: tm.primary.base,
    base: tm.primary.base,
    light: "rgba(2, 108, 223, 0.2)",
    muted: tm.primary.lightBase,
    restrained: "rgba(2, 108, 223, 0.4)",
    hover: tm.primary.medium,
    hoverLight: tm.primary.light,
    pressed: tm.primary.dark,
    pressedLight: tm.primary.muted
  },
  emerald: {
    ...tm.success,
    hover: tm.special.medium,
    pressed: tm.special.dark,
    muted: "rgba(27, 171, 30, 0.4)",
    restrained: "rgba(27, 171, 30, 0.2)"
  },
  outlined: {
    hover: "#ECF4FD",
    pressed: "#D9E8F9"
  },
  transparent: tm.transparent,
  summerSky: "#3AC7FF",
  blackPearl: tm.darkFill,
  slate: tm.gray02,
  moonstone: "#D1D1D1",
  shale: "#E6E6E6",
  quartz: tm.gray05,
  lightGray: "#E0E0E0",
  aquamarine: tm.accent04,
  turquoise: tm.accent03,
  ruby: {
    base: tm.error.base,
    darkRuby: tm.error.dark,
    lightRuby: tm.error.light
  },
  amber: {
    base: tm.caution.base,
    darkAmber: tm.caution.dark,
    lightAmber: tm.caution.light
  },
  moonrock: tm.gray03,
  diatomite: tm.gray04,
  heliotrope: {
    ...tm.accent01,
    muted: "#F3E1FF"
  },
  cruz: {
    ...tm.accent02,
    muted: "#e3cae3"
  },
  alert: {
    base: tm.error.base,
    dark: "#AE393A",
    light: "#EC9C9C",
    muted: "#F5CDCD"
  },
  caution: tm.caution,
  positive: {
    base: tm.success.base,
    dark: "#208E25",
    light: "#8DD58E",
    muted: "#C5E9C6"
  },
  defaultGradient: {
    from: tm.primary.base,
    to: "#3ac7ff"
  },
  // Onyx and white are specific for typography. Should not be removed
  onyx: tm.onyx,
  white: tm.white
};

export default colors;
