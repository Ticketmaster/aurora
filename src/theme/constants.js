const constants = {
  borderRadius: {
    small: "2px",
    large: "4px"
  },
  breakpoint: {
    small: "480px",
    medium: "768px",
    large: "1024px"
  },
  pillBorderRadius: "18px",
  thumbnailSize: "40px",
  drawer: {
    dismissAreaHeight: "60px",
    lineUpAreaHeight: "55px",
    desktopWidth: 26.05,
    mobileWidth: 80
  },
  device: {
    mobile: `(min-width: 480px)`,
    tablet: `(min-width: 768px)`,
    desktop: `(min-width: 1024px)`
  }
};

export default constants;
