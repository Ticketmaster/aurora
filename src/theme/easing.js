export const bezierEasy = "cubic-bezier(0.455, 0.03, 0.515, 0.955)";
export const bezierExit = "cubic-bezier(0.55, 0.085, 0.68, 0.53)";
