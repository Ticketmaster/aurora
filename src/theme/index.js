export { default as animations } from "./animations";
export { default as colors } from "./colors";
export { default as constants } from "./constants";
export { default as spacing } from "./spacing";
export { default as themes } from "./colorThemes";
export { default as typography } from "./typography";
export { default as zIndex } from "./zIndex";
