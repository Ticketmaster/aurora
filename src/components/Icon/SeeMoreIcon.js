import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";

import { spacing, constants } from "../../theme";
import Icon from "./";

const SeeMoreIconContainer = styled.div`
  padding: 0 ${spacing.large};
  display: none;
  align-items: center;
  justify-content: center;

  @media only screen and ${constants.device.mobile} {
    display: flex;
  }
`;

const SeeMoreIcon = ({ name, size, color, svgStyle }) => (
  <SeeMoreIconContainer>
    <Icon name={name} size={size} color={color} svgStyle={svgStyle} />
  </SeeMoreIconContainer>
);

SeeMoreIcon.defaultProps = {
  svgStyle: {}
};

SeeMoreIcon.propTypes = {
  name: PropTypes.string.isRequired,
  size: PropTypes.number.isRequired,
  color: PropTypes.string.isRequired,
  svgStyle: PropTypes.objectOf(PropTypes.any)
};

export default SeeMoreIcon;
