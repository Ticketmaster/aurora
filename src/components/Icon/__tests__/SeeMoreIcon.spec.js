import React from "react";
import renderer from "react-test-renderer";

import { colors } from "../../../theme";
import SeeMoreIcon from "../SeeMoreIcon";

describe("<SeeMoreIcon />", () => {
  it("renders SeeMoreIcon correctly", () => {
    const component = renderer.create(
      <SeeMoreIcon
        name="chevron-down"
        size={15}
        color={colors.cascade.light}
        desktopDisplay={false}
        mobileDisplay
        svgStyle={{ transform: "rotate(-90deg) " }}
      />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
