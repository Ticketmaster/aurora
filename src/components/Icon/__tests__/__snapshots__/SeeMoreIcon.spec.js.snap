// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SeeMoreIcon /> renders SeeMoreIcon correctly 1`] = `
.c0 {
  padding: 0 16px;
  display: none;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

@media only screen and (min-width:480px) {
  .c0 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
  }
}

<div
  className="c0"
>
  <svg
    height="15px"
    style={
      Object {
        "transform": "rotate(-90deg) ",
      }
    }
    viewBox="0 0 1024 1024"
    width="15px"
  >
    <path
      d="M1012.48 247.808c-15.053-15.053-40.96-15.053-56.013 0l-444.467 444.416-444.928-444.416c-15.053-15.002-41.062-14.438-54.784-0.614-4.454 3.789-7.68 8.448-9.728 13.824-1.69 4.608-2.56 9.523-2.56 14.49s0.87 9.933 2.56 14.592c2.048 5.376 5.274 10.035 8.96 13.21l472.73 472.73c7.526 7.526 17.101 11.52 27.75 11.52s20.224-3.942 27.75-11.469l472.73-472.73c7.526-7.526 11.52-17.101 11.52-27.75s-3.994-20.122-11.52-27.648z"
      fill="#b4b9bd"
    />
  </svg>
</div>
`;
